docs/man/alot.1
.\" DO NOT MODIFY THIS FILE!  It was generated by help2man 1.38.4.
.TH ALOT "1" "December 2011" "alot 0.21" "User Commands"
.SH NAME
mail user agent for the notmuch mail system
.SH SYNOPSIS
alot [-r] [-c CONFIGFILE] [-n NOTMUCHCONFIGFILE] [-C {1,16,256}]
     [-p DB_PATH] [-d {debug,info,warning,error}] [-l LOGFILE]
     [--version] [--help] [command]
.SH DESCRIPTION
A terminal-based interface for the notmuch mail system.
It features a modular and command prompt driven interface to provide a full MUA
experience as an alternative to the Emacs mode shipped with notmuch.

In all views, arrows, page-up/down, `j`,`k` and `Space` can be used to move the
focus.  `Escape` cancels prompts and `Enter` selects. Hit `:` at any time and type
in commands to the prompt. Usage information on any command can be listed by typing
`help YOURCOMMAND` to the prompt; The keybindings for the current mode are listet
upon pressing `?`.
.SH OPTIONS
.TP
\fB\-r\fR, \fB\-\-read\-only\fR
open db in read only mode
.TP
\fB\-c\fR, \fB\-\-config=\fR
config file [default: ~/.config/alot/config]
.TP
\fB\-n\fR, \fB\-\-notmuch\-config=\fR
notmuch config [default: ~/.notmuch\-config]
.TP
\fB\-C\fR, \fB\-\-colour\-mode=\fR
terminal colour mode [default: 256]. Must be 1, 16 or
256
.TP
\fB\-p\fR, \fB\-\-mailindex\-path=\fR
path to notmuch index
.TP
\fB\-d\fR, \fB\-\-debug\-level=\fR
debug log [default: info]. Must be one of
debug,info,warning or error
.TP
\fB\-l\fR, \fB\-\-logfile=\fR
logfile [default: /dev/null]
.HP
\fB\-\-version\fR
.TP
\fB\-\-help\fR
Display this help and exit.
.SS "Commands:"
.TP
search
start in a search buffer using the querystring provided as
parameter. See the SEARCH SYNTAX section of notmuch(1).
.TP
compose
compose a new message
.SH BUGS
Alot is a fairly new project, so expect misbehaviour. Bugs can be reported via the
issue tracker on the projecs github page (https://github.com/pazz/alot).
.SH AUTHOR
Patrick Totzke <patricktotzke at googlemail dot com>
.SH "SEE ALSO"
notmuch(1)


NEWS
current:

* completion for multiple recipients via AbooksCompleter
* header key completion for set/unset in envelope buffer
* generate and set a Message-ID header when constructing mails
* add sent and saved draft mails to the notmuch index and add custom tags
* optional highlighting of unread-tagged threads in search buffer
* case insensitive sorting of tagstrings in search buffer
* guess file encodings with libmagic
* theming of threadline depending on the presence of fixed tags
* new thread mode command: "remove" to delete messages from the index
* new thread mode command: "editnew" e.g. to continue drafts (bound to 'n')
* new envelope command: "save" to save as draft (bound to 'P')
* new thread mode command: "togglesource" to display raw messages (bound to 'h')



0.21

* avoid traceback infos from getting written on top of the ui
* new "--help" output, autogenerated manpage
* version string extracted from git for cli option "--version"
* command line subcommands: compose and search
* properly display multiple headers with the same key
* envelope.set option "--append"
* more detailed CUSTOMIZE docs
* multiple fixes for the envelope buffer
* exit on closing of last buffer is now optional
* die gracefully when config parsing fails
* random bugfixes in the ui
* determine attachments via the "Content-Disposition" header
* nicer alignment for messages in thread buffers
* deal with external commands as lists of strings instead of strings
* better filetype detection in attachments via magic numbers
* errbacks and correct calling of post-hooks for deferred commands
* add packaging info for debian
* envelope.headers stores lists of values for each key now
* default binding: 's' to 'toggletag unread' in search buffers


0.20

* extensive API docs
* fancy shortening for authors list
* moved default location for config to ~/.config/alot/config
* message templates
* hooks for pre/post editing and RE/FWD quotestrings
* recipient completion gives priority to abook of sender account
* smarter in-string-tab completion
* added ability to pipe messages/treads to custom shell commands
* initial command configurable in config file
* non-blocking prompt/choice (new syntax for prompts!)
* usage help for every command
* bindings help
* tons of fixes


0.11

This minor release is mostly bug fixes and some small features.
I wanted to release a more stable and usable version before I start
hacking on a thread view rewrite.

* renamed config section [tag translate] to [tag-translate]
* docs: more elaborate API docs, INSTALL and USAGE as markdown in github wiki
* more compact header displays in thread view
* command-line history (for one session)
* editor file encoding is now user configurable
* signatures for outgoing mails per account
* optional display of message content in search results
* config option for strftime formating of timestamps
* printing

* fix parse multiline headers from edited tempfile
* fix reply to unusually formated mails (e.g. no recipient)
* fix lots of encoding issues
* handle extra wide characters in tag widgets
* fixes in ui.prompt
* fix storing outgoing mails to sent_box
* more liberal header encoding for outgoing mails
* use mimetype lib to guess right content-type of attachments


alot.py
#!/usr/bin/python
"""
This file is part of alot, a terminal UI to notmuch mail (notmuchmail.org).
Copyright (C) 2011 Patrick Totzke <patricktotzke@gmail.com>

This program is free software: you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with this program.  If not, see <http://www.gnu.org/licenses/>.
"""
import argparse
import logging

from alot.db import DBManager
from alot.ui import UI


def parse_args():
    parser = argparse.ArgumentParser()
    parser.add_argument('-c', dest='colours',
                        type=int,
                        choices=[1,16,88,256],
                        default=16,
                        help='colour mode')
    parser.add_argument('-r', dest='read_only',
                        action='store_true',
                        help='open db in read only mode')
    parser.add_argument('-p', dest='db_path',
                        help='path to notmuch index')
    parser.add_argument('-d', dest='debug_level',
                        default='info',
                        choices=['DEBUG','INFO','WARNING','ERROR'],
                        help='debug level')
    parser.add_argument('-l', dest='logfile',
                        default='debug.log',
                        help='logfile')
    parser.add_argument('query', nargs='?',
                        default='tag:inbox AND NOT tag:killed',
                        help='initial searchstring')
    return parser.parse_args()


def main():
    args = parse_args()
    dbman = DBManager(path=args.db_path, ro=args.read_only)
    numeric_level = getattr(logging, args.debug_level.upper(), None)
    logging.basicConfig(level=numeric_level, filename=args.logfile)
    logger = logging.getLogger()
    ui = UI(db=dbman,
            log=logger,
            initialquery=args.query,
            colourmode = args.colours
    )

if __name__ == "__main__":
    main()
